nomos-services/blend/src/core/mod.rs

```diff
@@ -1054,37 +1064,43 @@ where
                             .1,
                         old_token_collector: old_session_blending_token_collector,
                         old_public_info: current_public_info,
-                        old_recovery_checkpoint: current_recovery_checkpoint,
                     };
                 }
             };
-            let state_updater = current_recovery_checkpoint.into_components().4;
+            let state_updater = current_recovery_checkpoint.into_components().6;
             let (new_scheduler, old_scheduler) = current_scheduler
                 .rotate_session(new_scheduler_session_info, settings.scheduler_settings());
             HandleSessionEventOutput::Transitioning {
                 new_crypto_processor: new_processor,
                 old_crypto_processor: current_cryptographic_processor,
                 new_scheduler,
                 old_scheduler,
-                new_token_collector: new_session_blending_token_collector,
-                old_token_collector: old_session_blending_token_collector,
+                new_token_collector: new_session_blending_token_collector.clone(),
```

Comment:
Also here, why do we need to return the collectors on their own and also as part of the recovery state?